src/core/environment.js

```diff
@@ -1406,6 +1406,102 @@ function environment(p5, fn){
           return new Vector(screenX, screenY, screenZ);
     }
   };
+  /**
+   * Converts 2D screen coordinates to 3D world coordinates.
+   *
+   * This function takes a vector and converts its coordinates
+   * from the world space to screen space. This can be useful for determining
+   * the mouse position relative to a 2D or 3D object.
+   *
+   * The Z component of the screen coordinates is treated as "depth", or
+   * distance from the camera, when converting to 3D world space.
+   *
+   * @method screenToWorld
+   * @param {p5.Vector} screenPosition The coordinates in screen space.
+   * @return {p5.Vector} A vector containing the 3D world space coordinates.
+   * @example
+   * <div>
+   * <code>
+   *
+   * function setup() {
+   *   createCanvas(100, 100);
+   *   describe('A rotating square with a line passing through the mouse drawn across it.');
+   * }
+   *
+   * function draw() {
+   *   background(220);
+   *
+   *   // Move to center and rotate
+   *   translate(width/2, height/2);
+   *   rotate(millis() / 1000);
+   *   rect(-30, -30, 60);
+   *
+   *   // Compute the location of the mouse in the coordinates of the square
+   *   let localMouse = screenToWorld(createVector(mouseX, mouseY));
+   *
+   *   // Draw a line parallel to the local Y axis, passing through the mouse
+   *   line(localMouse.x, -30, localMouse.x, 30);
+   * }
+   *
+   * </code>
+   * </div>
+   * @example
+   * <div>
+   * <code>
+   *
+   * function setup() {
+   *   createCanvas(100, 100, WEBGL);
+   *   describe('A rotating square with a line passing through the mouse drawn across it.');
+   * }
+   *
+   * function draw() {
+   *   background(220);
+   *
+   *   // Animate rotation
+   *   rotateZ(millis() / 1000);
+   *   rect(-30, -30, 60);
+   *
+   *   // Transform the 0, 0 point to get the depth of the square
+   *   let center = worldToScreen(createVector(0, 0))
+   *
+   *   // Compute the location of the mouse in the coordinates of the square,
+   *   // using the depth calculated earlier
+   *   let localMouse = screenToWorld(createVector(mouseX, mouseY, center.z));
+   *
+   *   // Draw a line parallel to the local Y axis, passing through the mouse
+   *   line(localMouse.x, -30, localMouse.x, 30);
+   * }
+   *
+   * </code>
+   * </div>
+   *
+   */
+  fn.screenToWorld = function(screenPosition) {
+    const renderer = this._renderer;
+    if (renderer.drawingContext instanceof CanvasRenderingContext2D) {
+      // Handle 2D context
+      const inverseTransformMatrix = new DOMMatrix()
+        .multiply(renderer.drawingContext.getTransform().inverse())
+        .scale(renderer._pInst.pixelDensity());
```

Comment:
This code could potentially be moved to a `Renderer2D.calculateCombinedMatrix()` function, which would let us get rid of the `if` above.
Thoughts on such a refactoring (I would do the same to `worldToScreen`)?

Reply:
That makes sense to me if we want to extract these into functions on the respective renderers. `calculateCombinedMatrix` is specifically for model-view currently, so maybe we could make a new function for it that returns a matrix going right from p5 screen coordinates --> world coordinates?

Reply:
`calculateScreenToWorldMatrix`? :sweat_smile: 

Also, do note that `calculateCombinedMatrix` is used in only 3 places right now. One is internally in RendererGL, one is in screenToWorld/worldToScreen, and one is in accessibility/outputs.js's `_getPos` function (since your 631a14d157b3a066be2e941dc1248f8ab858e829), which... looks like it's almost exactly the same thing as `worldToScreen` and, if I'm not wrong, will be fine (better!) with the projection matrix applied as well?

Reply:
Right, there's probably more use for the fully combined one, but there are some specific spots (passing it into `uMVMatrix` for shaders mostly) where we need just model + view and nothing else, so as long as we keep that the same, it's all good! we can rename the existing one to something else if we need.

Reply:
@davepagurek I switched it to a `getWorldToScreenMatrix` function that carries out the whole calculation, and changed the `_getPos` in accessibility/outputs.js to use it instead (in fact, going through worldToScreen directly).
Let me know if you feel like a `RendererGL.getModelViewMatrix()` function would be useful.